app/jobs/hyku_addons/reindex_model_job.rb

```diff
@@ -44,8 +43,11 @@ def mint_doi(work)
         work.update(doi: [register_doi.identifier])
       end
 
-      def can_mint_for?(work)
-        work.doi.present? || work.visibility != "open" || workflow_state(work)&.workflow_state_name != "deposited"
+      def cant_mint_for(work)
+        state = workflow_state(work)
+        cannot_use_workflow_state = ["deposited", nil].exclude?(state&.workflow_state_name)
+
+        !(work.creator.blank? || work.doi.present? || work.visibility != "open" || cannot_use_workflow_state)
```

Comment:
The method is named negatively, and this condition at the end is negated.  If this actually works please just name it positively, and remove the !.